org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java

```diff
@@ -600,4 +636,64 @@ private void applyToNode(AbstractElement node, String actionText, MultiTextEdit
 				});
 	}
 
+	@Fix(ILLEGAL_RULE_REFERENCE)
+	public void fixConvertTerminalFragmentToTerminalRule(Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, //
+				"Convert terminal fragment to terminal rule", //
+				"Convert terminal fragment to terminal rule", //
+				NULL_QUICKFIX_IMAGE, //
+				(context) -> {
+					IXtextDocument xtextDocument = context.getXtextDocument();
+					xtextDocument.modify(new IUnitOfWork.Void<XtextResource>() {
+
+						@Override
+						public void process(XtextResource state) throws Exception {
+							if (state == null) {
+								return;
+							}
+
+							XtextGrammarAccess grammarAccess = state.getResourceServiceProvider().get(XtextGrammarAccess.class);
+							RuleCall ruleCall = (RuleCall) state.getEObject(issue.getUriToProblem().fragment());
+							ICompositeNode ruleNode = NodeModelUtils.findActualNodeFor(ruleCall.getRule());
+							for (INode node : ruleNode.getAsTreeIterable()) {
+								if (node.getGrammarElement() != null && node.getGrammarElement() == grammarAccess.getTerminalRuleAccess()
+										.getFragmentFragmentKeyword_2_0_0_0()) {
+									Region fragmentRegion = new Region(node.getOffset(), node.getLength());
+									xtextDocument.replace(fragmentRegion.getOffset(), fragmentRegion.getLength() + 1, "");
+									break;
+								}
+							}
+						}
+					});
+				});
+	}
+
+	@Fix(XtextConfigurableIssueCodes.INVALID_HIDDEN_TOKEN_FRAGMENT)
+	public void fixInvalidHiddenTokenFragment(Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, //
```

Comment:
Please extract the common parts into an own method.
I'm not terribly sure, but I think you can put multiple `@Fix` on the same method.